Azure/sonic-mgmt
# Convert 'read_mac' ansible test to pytest

Signed-off-by: Andriy Yurkiv ayurkiv@nvidia.com
### Description of PR



Summary:
Fixes # (issue)

### Type of change




### Approach
#### What is the motivation for this PR?
Migrating tests from ansible to pytest framework

#### How did you do it?
Converted read_mac ansible test to pytest.
Old ansible test :https://github.com/Azure/sonic-mgmt/blob/master/ansible/roles/test/tasks/read_mac_metadata.yml 
#### How did you verify/test it?
```
py.test read_mac/test_read_mac_metadata.py --iteration <iter> --image1 <image1URL> --image2 <image2URL> --minigraph1 <file1> --minigraph2 <file2> --inventory ../ansible/inventory --host-pattern <host> --module-path ../ansible/library/  --testbed <host>-<topology> --testbed_file ../ansible/testbed.csv --show-capture=no --capture=no --log-cli-level info -ra
```
iteration, image1 and image2 params are mandatory 
minigraph1  and minigraph2 are optional

#### Any platform specific information?
N/A
#### Supported testbed topology if it's a new test case?
any
### Documentation 
'read mac' test  flow:
- Download sonic  image
- Install it
- Reboot
- Apply new minigraph (optional)
- Check if MAC is correct on interfaces
- Check MTU on interfaces
- Repeat N times with different images

**Note**
This test has at least 2 parameters that cannot  be defined as default (**image1** and **image2**)
Because of this backward compatibility currently  does not work (I mean ability to run test using ansible-playbook)
```
ansible-playbook test_sonic.yml -i inventory --limit {DUT_NAME} -e testbed_name={TESTBED_NAME} -e testbed_type={TESTBED_TYPE} -e testcase_name=read_mac -e iterations={ITERATIONS} -e image1={IMAGE1} -e image2={IMAGE2}
```
Current implementation of ansible  to pytest wrapper does not allow to pass python input params. And this may be motivation to improve it fuctionality in future
